backend/svirt.pm

```diff
@@ -116,7 +116,8 @@ sub run_cmd {
         username => 'root'
     );
     my $chan = $self->{ssh}->channel();
-    $chan->exec($cmd);
+    $chan->exec($cmd)
+      or $chan->die_with_error;
```

Comment:
Have you ever actually tested this? Considering the documentation it looks wrong. The `die_with_error` should be called on the `Net::SSH2` object and not on the channel.

---

But I'm going to improve the error handling for https://progress.opensuse.org/issues/44771 anyways. There are also more occurrences like the creation of the channel object itself which need to be covered.

Reply:
Yes, I tested this. Could be that I had that fixed already locally or maybe I did not run into this yet. Thanks in general for taking over and bringing it to a good end.